rules/typescript.js

```diff
@@ -0,0 +1,68 @@
+// eslint-disable-next-line import/no-commonjs
+module.exports = {
+  plugins: ['@typescript-eslint'],
+  rules: {
+    '@typescript-eslint/adjacent-overload-signatures': ['error'],
+    '@typescript-eslint/array-type': ['error', 'array-simple'],
+    '@typescript-eslint/ban-types': ['error'],
+    '@typescript-eslint/ban-ts-ignore': ['off'],
+    '@typescript-eslint/camelcase': ['error'],
+    '@typescript-eslint/class-name-casing': ['error'],
+    '@typescript-eslint/explicit-function-return-type': ['off'],
```

Comment:
Depends on our usage but it might be interesting to enable it with both option enable.

> https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/explicit-function-return-type.md#options

Reply:
No objection on this. There will be 7 errors to fix in InstantSearch.js but it should be pretty straightforward.